html/includes/hostbox-menu.inc.php

```diff
@@ -27,18 +27,38 @@
             </div>';
     }
 
-    echo '</div>
-        <div class="row">
-        <div class="col-xs-1">
-        <a href="telnet://'.$device['hostname'].'"><i class="fa fa-terminal fa-lg icon-theme"  title="Telnet to ' . $device['hostname'] . '"></a>
-        </div>
-        <div class="col-xs-1">
-        <a href="ssh://'.$device['hostname'].'"><i class="fa fa-lock fa-lg icon-theme"  title="SSH to ' . $device['hostname'] . '"></a>
-        </div>
-        <div class="col-xs-1">
-        <a href="https://' . $device['hostname'] . '" target="_blank" rel="noopener"><i class="fa fa-globe fa-lg icon-theme"  title="Launch browser https://' . $device['hostname'] . '"></i></a>
-        </div>
-        </div>';
+    if (isset($config['gateone']['server'])) {
```

Comment:
This contains a lot of duplication and you should just do the `if()` checks around the actual lines that need to be different.